front/src/js/service.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * @author Yannis Malgorn <yannismalgorn@gmail.com>
+ * @see {@link https://github.com/mdl29/scratchy|GitHub}
+ * @requires module:axios/axios
+ */
+/*
+- add axios cdn link in index.html :
+     	<script src="https://unpkg.com/axios@0.21.1/dist/axios.min.js"></script>
+- call service.js in index.html :
+    <script src="/front/js/service.js"></script>
+- start scratchy service : 
+    var api = new ScratchyService("http://localhost:5000/api");
+    api;
+- call a function :
+    api.function();|
+*/
+/**
+ * 
+ * @constructor
+ * @param {string} apiUrl - api url , eg: http://localhost:5000/api
+ */
+function ScratchyService(apiUrl) {
+    let _this = this;
+    this.cacheUsers = []; // List of users, used as cache
+
+//                               ROOM FUNCTIONS
+
+
+    /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @typedef {Object} Room
```

Comment:
@typedef should be defined somewhere else outsite of the `ScratchyService` declaration.

See my comment here : https://github.com/mdl29/scratchy/pull/152#discussion_r625215300 

You need to separe @typedef comment block from method comment block.